app/src/main/kotlin/com/adesso/movee/scene/profile/ProfileViewModel.kt

```diff
@@ -58,12 +54,8 @@ class ProfileViewModel @Inject constructor(
     }
 
     private fun fetchUserDetails() {
-        bgScope.launch {
-            val userDetailsResult = fetchUserDetailsUseCase.run(UseCase.None)
-
-            onUIThread {
-                userDetailsResult.either(::handleFailure, ::postUserDetails)
-            }
+        fetchUserDetailsUseCase.invoke(viewModelScope, UseCase.None) {
+            it.either(::handleFailure, ::postUserDetails)
```

Comment:
postUserDetails() uses LiveData's setValue() function which must be called from main thread. Either we need to use postValue(), or switch context when calling postUserDetails()